crypto/crypto.go

```diff
@@ -0,0 +1,36 @@
+package crypto
+
+import (
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/sha3"
+)
+
+const (
+	HashSizeByte = 32
+	HashID       = "SHAKE128"
+)
+
+func Digest(ms ...[]byte) []byte {
+	h := sha3.NewShake256()
+	for _, m := range ms {
+		h.Write(m)
+	}
+	var ret [HashSizeByte]byte
+	h.Read(ret[:])
+	return ret[:]
```

Comment:
I think this is clearer as,

	ret := make([]byte, HashSizeByte)
	h.Read(ret)
	return ret